contentcuration/contentcuration/db/models/manager.py

```diff
@@ -81,55 +83,43 @@ def _get_next_tree_id(self, *args, **kwargs):
         return new_id
 
     @contextlib.contextmanager
-    def _attempt_lock(self, tree_ids, values):
+    def _attempt_lock(self, tree_ids, shared_tree_ids=None):
         """
         Internal method to allow the lock_mptt method to do retries in case of deadlocks
         """
+        shared_tree_ids = shared_tree_ids or []
+
         start = time.time()
         with transaction.atomic():
             # Issue a separate lock on each tree_id
             # in a predictable order.
             # This will mean that every process acquires locks in the same order
             # and should help to minimize deadlocks
             for tree_id in tree_ids:
-                execute_queryset_without_results(
```

Comment:
Is there any scenario where not applying the row level locks when there is an update operation happening might be harmful? I'm assuming that the worst that would happen is a deadlock if multiple update operations were done in a single transaction?

Reply:
I just did a quick check of the concurrency test case here with your PR locally: https://github.com/learningequality/studio/blob/hotfixes/contentcuration/contentcuration/tests/viewsets/test_contentnode.py#L58

And it produced a failure, meaning that we had a tree corruption as a result of concurrent tree operations. Could you try replicating locally too? Might mean that we have to go for a 'double lock' for safety. But maybe we can do the advisory locks in a loop first, and when we get past them, in a second loop we do the write locks.

Reply:
I'll take a look, thanks!

Reply:
The issue appeared to be that I made the mistake of calling the context manager function, without a context so it had no effect.